providers/task.rb

```diff
@@ -41,7 +41,7 @@
     options['ST'] = @new_resource.start_time unless @new_resource.start_time.nil?
     options['TR'] = "\"#{@new_resource.command}\" "
     options['RU'] = @new_resource.user
-    options['RP'] = @new_resource.password if @new_resource.password
+    options['RP'] = password if password
```

Comment:
"password if password" sounds like we're setting the password if we have one. it'd be nice to change the name of the method to something that conveys more of its logic.